### PR TITLE
Publish a shadowed uber jar of Arrow

### PR DESCRIPTION
**Pull request by raulraja**
The uber jar contains relocation packages so it may be consumed by `arrow-meta`.
Uses Arrow previous version to build Arrow and other `arrow-meta` dependents.

### CODE REVIEW

File: modules/shadow/arrow-shadow/build.gradle [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    dependencies {
+        classpath 'com.github.jengelman.gradle.plugins:shadow:4.0.0'
+    }
+}
+
+apply plugin: 'com.github.johnrengelman.shadow'
+apply plugin: 'java'
+
+ext {
+    metaArrowVersion = '0.7.3'
```
Comment (pakoito) [inline modules/shadow/arrow-shadow/build.gradle]: Should this be on the root build.gradle instead?

Comment (raulraja) [reply modules/shadow/arrow-shadow/build.gradle]: yeah, I'll move it there as part of the `arrow-meta` uber PR

**Comment by raulraja** [comment]
@pakoito what do we do in cases where codecov BS shows up decreasing like this even if there are no code changes?

**Comment by pakoito** [comment]
@raulraja we always merge regardless of codecov, AFAIK. It's a metric, but not one we follow hahah

**Comment by raulraja** [comment]
LOL, I like that 🔥

**Pull request merged**